From 68f7d9cb901667c979202f5ce0542b2524cb0d27 Mon Sep 17 00:00:00 2001
From: Baisong Zhong <zhongbaisong@huawei.com>
Date: Fri, 30 Sep 2022 11:04:27 +0800
Subject: [PATCH 0245/1076] [Huawei] Huawei BMA: fix return value check of
 kmalloc()

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187729

--------------------------------

Fix the following eno.cocci and null_return_check.cocci warning.
In case of error, kmalloc() is checked for NULL. We should not
use IS_ERR() to check the return value again, so remove it.

./drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c:435:23-29:
ERROR: allocation function on line 434 returns NULL not ERR_PTR on failure
./drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c:307:5-11:
ERROR: allocation function on line 305 returns NULL not ERR_PTR on failure
./drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c:317:5-11:
ERROR: allocation function on line 315 returns NULL not ERR_PTR on failure
./drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c:138:5-11:
ERROR: allocation function on line 137 returns NULL not ERR_PTR on failure
./drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c:147:5-11:
ERROR: allocation function on line 146 returns NULL not ERR_PTR on failure

Signed-off-by: Baisong Zhong <zhongbaisong@huawei.com>
---
 drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c  | 4 ++--
 drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c | 4 ++--
 drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c
index 0c17cd2bae49..2b142ae9bff6 100644
--- a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c
+++ b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_panic.c
@@ -135,7 +135,7 @@ int kbox_panic_init(void)
 	int ret = KBOX_TRUE;
 
 	g_panic_info_buf = kmalloc(SLOT_LENGTH, GFP_KERNEL);
-	if (IS_ERR(g_panic_info_buf) || !g_panic_info_buf) {
+	if (!g_panic_info_buf) {
 		KBOX_MSG("kmalloc g_panic_info_buf fail!\n");
 		ret = -ENOMEM;
 		goto fail;
@@ -144,7 +144,7 @@ int kbox_panic_init(void)
 	memset(g_panic_info_buf, 0, SLOT_LENGTH);
 
 	g_panic_info_buf_tmp = kmalloc(SLOT_LENGTH, GFP_KERNEL);
-	if (IS_ERR(g_panic_info_buf_tmp) || !g_panic_info_buf_tmp) {
+	if (!g_panic_info_buf_tmp) {
 		KBOX_MSG("kmalloc g_panic_info_buf_tmp fail!\n");
 		ret = -ENOMEM;
 		goto fail;
diff --git a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c
index 3b04ba206108..630a1e16ea24 100644
--- a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c
+++ b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_printk.c
@@ -304,7 +304,7 @@ int kbox_printk_init(int kbox_proc_exist)
 
 	g_printk_info_buf = kmalloc(SECTION_PRINTK_LEN,
 				    GFP_KERNEL);
-	if (IS_ERR(g_printk_info_buf) || !g_printk_info_buf) {
+	if (!g_printk_info_buf) {
 		KBOX_MSG("kmalloc g_printk_info_buf fail!\n");
 		ret = -ENOMEM;
 		goto fail;
@@ -314,7 +314,7 @@ int kbox_printk_init(int kbox_proc_exist)
 
 	g_printk_info_buf_tmp = kmalloc(SECTION_PRINTK_LEN,
 					GFP_KERNEL);
-	if (IS_ERR(g_printk_info_buf_tmp) || !g_printk_info_buf_tmp) {
+	if (!g_printk_info_buf_tmp) {
 		KBOX_MSG("kmalloc g_printk_info_buf_tmp fail!\n");
 		ret = -ENOMEM;
 		goto fail;
diff --git a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c
index 49690bab1cef..9f6dfe55e3fb 100644
--- a/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c
+++ b/drivers/net/ethernet/huawei/bma/kbox_drv/kbox_ram_op.c
@@ -432,7 +432,7 @@ int kbox_write_op(long long offset, unsigned int count,
 		return KBOX_FALSE;
 
 	temp_buf_char = kmalloc(TEMP_BUF_DATA_SIZE, GFP_KERNEL);
-	if (!temp_buf_char || IS_ERR(temp_buf_char)) {
+	if (!temp_buf_char) {
 		KBOX_MSG("kmalloc temp_buf_char fail!\n");
 		up(&user_sem);
 		return -ENOMEM;
-- 
2.17.1

